### ISSUE TITLE
Extreme VDX not properly disabling output paging (TACACS+ not allowing it)

### ISSUE THREAD

**Issue by doug1as** [ktbyers/netmiko#1582]
Hello everyone,

I'am trying access one Brocade VDX with NOS firmware, and run some commands, i tried in Device Type:
'device_type': 'extreme_vdx'
'device_type': 'brocade_nos'
'device_type': 'brocade_vdx'
'device_type': 'brocade'

Some commands works, like show version.

But some sommands doesn't work, like show running-config, show vlan brief
I always receive this message when i try:

Traceback (most recent call last):
  File "base.py", line 32, in <module>
    output = host_connect.send_command("show runn")
  File "/usr/local/lib/python3.5/dist-packages/netmiko/base_connection.py", line 1337, in send_command
    search_pattern
OSError: Search pattern never detected in send_command_expect: L \ - BR \ -PR \ -CTA \ -PV \ -SCR \ -02 \ #

**Comment by paulhenry10** [comment]
Hi,

I think I ran on the same problem with when trying to access Cisco switch specifically 4500 series.

Please see error below:
Traceback (most recent call last):
  File "everyday_backup.py", line 36, in <module>
    output = net_connect.send_command(command_string=command, strip_prompt=False, strip_command=False)
  File "C:\Users\pmcagayan\AppData\Local\Programs\Python\Python38\lib\site-packages\netmiko\base_connection.py", line 1335, in send_command
    raise IOError(
OSError: Search pattern never detected in send_command_expect: P3\.CAFOLC4507\-20\.222\#

It only occurs on Cisco 4507. I do not have any problem on Cisco 2960 series switch.

This is working perfectly before so I think an update bug/problem?

**Comment by ktbyers** [comment]
@doug1as Which version of Netmiko are you using?

**Comment by ktbyers** [comment]
@paulhenry10 I think that is likely a different issue--please open a separate GitHub issue. The backslash is just an indicator that it is a regular expression pattern and that special characters are being escaped.

**Comment by doug1as** [comment]
@ktbyers i'am using this version with python3.5:

Name: netmiko
Version: 2.4.2
Summary: Multi-vendor library to simplify Paramiko SSH connections to network devices
Home-page: https://github.com/ktbyers/netmiko
Author: Kirk Byers
Author-email: ktbyers@twb-tech.com
License: MIT
Location: /usr/local/lib/python3.5/dist-packages

Regards.

**Comment by ktbyers** [comment]
You probably need to be using the latest version of Netmiko (3.0.0) which requires Python 3.6.

Have you tried it on Netmiko 3.0?

**Comment by doug1as** [comment]
Not yet, i will try and inform you, i will be able to test tomorrow.

Thanks a lot ...

**Comment by doug1as** [comment]
@ktbyers i tried with netmiko version 3.0.0, but the error still occur:

Traceback (most recent call last):
  File "base.py", line 32, in <module>
    output = host_connect.send_command("show vlan brief")
  File "/usr/local/lib/python3.7/dist-packages/netmiko/base_connection.py", line 1407, in send_command
    search_pattern
OSError: Search pattern never detected in send_command_expect: L\ -BR\ -PR\ -CTA\ -PV\ -SCR\ -02\ #


Name: netmiko
Version: 3.0.0
Summary: Multi-vendor library to simplify Paramiko SSH connections to network devices
Home-page: https://github.com/ktbyers/netmiko
Author: Kirk Byers
Author-email: ktbyers@twb-tech.com
License: MIT
Location: /usr/local/lib/python3.7/dist-packages
Requires: setuptools, scp, paramiko, textfsm, pyserial
Required-by: 

Name: setuptools
Version: 40.8.0
Summary: Easily download, build, install, upgrade, and uninstall Python packages
Home-page: https://github.com/pypa/setuptools
Author: Python Packaging Authority
Author-email: distutils-sig@python.org
License: UNKNOWN
Location: /usr/lib/python3/dist-packages
Requires: 
Required-by: netmiko

Name: scp
Version: 0.13.2
Summary: scp module for paramiko
Home-page: https://github.com/jbardin/scp.py
Author: James Bardin
Author-email: j.bardin@gmail.com
License: LGPL
Location: /usr/local/lib/python3.7/dist-packages
Requires: paramiko
Required-by: netmiko

Name: paramiko
Version: 2.7.1
Summary: SSH2 protocol library
Home-page: https://github.com/paramiko/paramiko/
Author: Jeff Forcier
Author-email: jeff@bitprophet.org
License: LGPL
Location: /usr/local/lib/python3.7/dist-packages
Requires: cryptography, bcrypt, pynacl
Required-by: scp, netmiko

Name: textfsm
Version: 1.1.0
Summary: Python module for parsing semi-structured text into python tables.
Home-page: https://github.com/google/textfsm
Author: None
Author-email: None
License: Apache License, Version 2.0
Location: /usr/local/lib/python3.7/dist-packages
Requires: future, six
Required-by: netmiko

Name: pyserial
Version: 3.4
Summary: Python Serial Port Extension
Home-page: https://github.com/pyserial/pyserial
Author: Chris Liechti
Author-email: cliechti@gmx.net
License: BSD
Location: /usr/local/lib/python3.7/dist-packages
Requires: 
Required-by: netmiko

**Comment by ktbyers** [comment]
@paulhenry10 Can you include the relevant Python code?

Also you probably need to turn on logging and include the logging output (as there isn't really enough information in the output to determine what is occurring).

https://github.com/ktbyers/netmiko/blob/develop/COMMON_ISSUES.md#enable-netmiko-logging-of-all-reads-and-writes-of-the-communications-channel

**Comment by paulhenry10** [comment]
> @paulhenry10 Can you include the relevant Python code?
> 
> Also you probably need to turn on logging and include the logging output (as there isn't really enough information in the output to determine what is occurring).
> 
> https://github.com/ktbyers/netmiko/blob/develop/COMMON_ISSUES.md#enable-netmiko-logging-of-all-reads-and-writes-of-the-communications-channel

Hi sir Kirk,

I opened new thread and this is now okay on my side. Kindly refer to below link 

https://github.com/ktbyers/netmiko/issues/1584#issuecomment-588600182

**Comment by ktbyers** [comment]
@doug1as Can you include the full exception stack trace?

**Comment by doug1as** [comment]
@ktbyers this is the output

Traceback (most recent call last):
  File "go.py", line 31, in <module>
    go_vendors.all_devices(username, password, hostname, ip, device_type, vendor, row_vendor)
  File "/home/a0094668/scripts/go_vendors.py", line 29, in all_devices
    output = host_connect.send_command(row[row_vendor])
  File "/usr/local/lib/python3.7/dist-packages/netmiko/base_connection.py", line 1407, in send_command
    search_pattern
OSError: Search pattern never detected in send_command_expect: L\ -BR\ -RS\ -CAN\ -AU\ -SCR\ -01 \#

**Comment by ktbyers** [comment]
Which driver are you using (i.e. which Netmiko device_type)?

The command is prompting you for:

```
�[7m--More--�[27m
```

Which shouldn't happen with the right driver (of course could be bugs in the code also).

**Comment by ktbyers** [comment]
You should be using one of these:

```
'device_type': 'extreme_vdx'
'device_type': 'brocade_nos'
'device_type': 'brocade_vdx'
```

They are the same driver.

**Comment by doug1as** [comment]
@ktbyers in this last post i used brocade_nos, but I tried with others drivers too, all of them has the same behavior.

**Comment by ktbyers** [comment]
Hmmm, strange...I don't see any "paging disable" in this driver (which makes me wonder if it was never implemented correctly).

What is the command to turn off the paging of output on the VDX?

**Comment by doug1as** [comment]
@ktbyers the command is terminal length 0, cisco like.
I tried but my user don't have this permission on TACACs profile rs.

FYI: I'm having the same problem with driver ruckus_fastiron, in our ICX7450 with FastIron OS, the behavior/stack trace is tha same.

**Comment by ktbyers** [comment]
Okay, this is a bug that will need fixed. Netmiko should automatically disable this after establishing the connection.

**Comment by doug1as** [comment]
@ktbyers for driver ruckus_fastiron, I need open new issue? or can you work with same informations ? is the same issue...

**Comment by ktbyers** [comment]
Does `fastiron` use terminal length 0 also?

**Comment by doug1as** [comment]
@ktbyers in fastiron is: **skip-page-display**

**Comment by ktbyers** [comment]
@doug1as Looking at the Netmiko code some more and at your logging output. Netmiko is sending "terminal length 0". It looks to me like the VDX is responding with:

```
# Response to "terminal length 0" command
DEBUG:netmiko:read_channel: -----------------------^
syntax error: unknown argument.
```

The most likely cause of this is what you described here:
```
I tried but my user don't have this permission on TACACs profile rs.
```

So I think that is probably an issue on your side. Let me know if you have facts that are inconsistent with that.

I still need to look into the FastIron.

**Comment by ktbyers** [comment]
Okay, the fastiron driver is also disabling paging:

```
class RuckusFastironBase(CiscoSSHConnection):
    """Ruckus FastIron aka ICX support."""

    def session_preparation(self):
        """FastIron requires to be enable mode to disable paging."""
        self._test_channel_read()
        self.set_base_prompt()
        self.enable()
        self.disable_paging(command="skip-page-display")
        # Clear the read buffer
        time.sleep(0.3 * self.global_delay_factor)
        self.clear_buffer()
```

So we would need to troubleshoot that more to see what is going on there. Let's focus on the VDX issue for now.

**Comment by doug1as** [comment]
@ktbyers, hummm so i need to try in one i have these permission, i'll try and inform you about this... Thanks a lot ...

**Comment by doug1as** [comment]
@ktbyers you are correct, i tryed commands with **| exclude nomore**, like:
**show run | exclude nomore**

and now works fine...


Thanks a lot for your support and your time !!!
